L1Trigger/CSCTriggerPrimitives/src/CSCGEMTriggerLUTGenerator.cc

```diff
@@ -0,0 +1,311 @@
+#include "L1Trigger/CSCTriggerPrimitives/src/CSCGEMTriggerLUTGenerator.h"
+#include "Geometry/GEMGeometry/interface/GEMEtaPartitionSpecs.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+
+void CSCGEMTriggerLUTGenerator::generateLUTsME11(unsigned theEndcap, unsigned theStation, unsigned theSector, unsigned theSubsector, unsigned theTrigChamber)
+{
+  bool gemGeometryAvailable(false);
+  if (gem_g != nullptr) {
+    LogTrace("CSCGEMTriggerLUTGenerator")
+      << "+++ generateLUTsME11() called for ME11 chamber! +++ \n";
+    gemGeometryAvailable = true;
+  }
+  
+  // CSC geometry
+  CSCTriggerGeomManager* geo_manager(CSCTriggerGeometry::get());
+  const CSCChamber* cscChamberME1b(geo_manager->chamber(theEndcap, theStation, theSector, theSubsector, theTrigChamber));
+  const CSCDetId me1bId(cscChamberME1b->id());
+  const CSCDetId me1aId(me1bId.endcap(), 1, 4, me1bId.chamber());
+  const CSCChamber* cscChamberME1a(csc_g->chamber(me1aId));
+  
+  // check for GEM geometry
+  if (not gemGeometryAvailable){
+    LogTrace("CSCGEMTriggerLUTGenerator")
+      << "+++ generateLUTsME11() called for ME11 chamber without valid GEM geometry! +++ \n";
+    return;
+  }
+  
+  // CSC trigger geometry
+  const CSCLayer* keyLayerME1b(cscChamberME1b->layer(3));
+  const CSCLayerGeometry* keyLayerGeometryME1b(keyLayerME1b->geometry());
+  const CSCLayer* keyLayerME1a(cscChamberME1a->layer(3));
+  const CSCLayerGeometry* keyLayerGeometryME1a(keyLayerME1a->geometry());
+  
+  const int region((theEndcap == 1) ? 1: -1);
+  const GEMDetId gem_id(region, 1, theStation, 1, me1bId.chamber(), 0);
+  const GEMChamber* gemChamber(gem_g->chamber(gem_id));
+  
+  // LUT<roll,<etaMin,etaMax> >    
+  std::map<int,std::pair<double,double> > gemRollToEtaLimits_;
+
+  int n = 1;
+  for(auto roll : gemChamber->etaPartitions()) {
+    const float half_striplength(roll->specs()->specificTopology().stripLength()/2.);
+    const LocalPoint lp_top(0., half_striplength, 0.);
+    const LocalPoint lp_bottom(0., -half_striplength, 0.);
+    const GlobalPoint gp_top(roll->toGlobal(lp_top));
+    const GlobalPoint gp_bottom(roll->toGlobal(lp_bottom));
+    gemRollToEtaLimits_[n] = std::make_pair(gp_top.eta(), gp_bottom.eta());
+    ++n;
+  }
+
+  LogTrace("CSCGEMTriggerLUTGenerator")
+    << "GEM roll to eta limits" << std::endl;
+  for(auto p : gemRollToEtaLimits_) {
+    LogTrace("CSCGEMTriggerLUTGenerator") << "{"<< p.first << ", {" << (p.second).first << ", " << (p.second).second << "}}, " << std::endl;
+  }
+  
+  // LUT<WG,<etaMin,etaMax> >
+  std::map<int,std::pair<double,double> > cscWGToEtaLimits_;
```

Comment:
Hi @dildick - make this a vector (or explain why it shouldn't be - its indexed from 1->N both times its defined)